rsconf/component/devbox.py

```diff
@@ -53,10 +53,11 @@ def internal_build_write(self):
                 [z.run_d, z.run_d, "ro"],
             ],
         )
+        u = jc.devbox.users[self.user_name]
         systemd.docker_unit_enable(
             self,
             jc,
-            image=jc.devbox.docker_image,
+            image=u.docker_image if isinstance(u, dict) else jc.devbox.docker_image,
```

Comment:
This pattern repeats. Compile is for handling this: converting arbitrary input formats into canonical form with defaults filled in. Put self.docker_image, defaulting to devbox.docker_image.

Reply:
d20cddb